src/rtc_pl031.rs

```diff
@@ -135,6 +135,18 @@ pub struct RTC<EV: RTCEvents> {
     events: EV,
 }
 
+fn get_epoch_time() -> u32 {
+    let epoch_time = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        // This expect should never fail because UNIX_EPOCH is in 1970,
+        // and the only possible failure is if `now` time is before UNIX EPOCH.
+        .expect("SystemTime::duration_since failed");
+    // The following conversation is save because u32::MAX would correspond to
+    // year 2106, and the RTC works exclusively with 32-bit registers, so we
+    // wouldn't be able to fit more than this in the LR (Load Register).
+    epoch_time.as_secs() as u32
```

Comment:
Hmm, it's still not clear to me why this conversion is safe. If we'd be in 2107, wouldn't as_secs() be > u32::MAX? Should we simplify this comment saying we don't expect to use this after 2106 or am I missing something?

Reply:
Epoch time would only not fit in the u32 if we were in year 2106. Since that is a reaaally long time from now, this conversion is safe. We would also not be able to use RTC at that point (at least this one), because the register that holds this value is an u32.

Reply:
Can we rephrase it to something like:

```
    // The following conversion is safe because u32::MAX would correspond to
    // year 2106, and we don't expect to use it afterwards. We are doing this
    // conversion because the RTC works exclusively with 32-bit registers, so we
    // wouldn't be able to fit more than this in the LR (Load Register).
```
?
The way I was reading the comment, the fact that the RTC works with 32bit registers is a reason why the conversion is safe, and if I get things right, this is not true.

Reply:
I've updated it. Let me know if it makes more sense now.